### ISSUE TITLE
Wrong error message for using the wrong macro "style" for the autodiff macro

### ISSUE THREAD

**Issue by sYgbpNA5UEVufjsHNwF3Xvf9ANZi5bNs** [EnzymeAD/rust#185]
`rustc +enzyme -vV` output:

rustc 1.84.0-nightly (1203575c6 2024-10-30)
binary: rustc
commit-hash: 1203575c627a2e96bc3543145526b5091a58be65
commit-date: 2024-10-30
host: x86_64-unknown-linux-gnu
release: 1.84.0-nightly
LLVM version: 19.1.1

example:
```rust
#![feature(autodiff)]
use std::autodiff::autodiff;

#[autodiff = "Forward"]
fn f(x: f32) -> f32 {
    x * x
}
#[autodiff]
fn g(x: f32) -> f32 {
    x * x
}

fn main() {}
```
complilation error:
```
error: autodiff must be applied to function
 --> src/main.rs:5:1
  |
5 | / fn f(x: f32) -> f32 {
6 | |     x * x
7 | | }
  | |_^

error: autodiff must be applied to function
  --> src/main.rs:9:1
   |
9  | / fn g(x: f32) -> f32 {
10 | |     x * x
11 | | }
   | |_^

error: could not compile `enzyme-test` (bin "enzyme-test") due to 2 previous errors

```

while both are wrong the error message stating the macro must be applied to functions only doesn't make sense here.

I don't know how the macro styles like `#[foo]` as opposed to `#[foo(Bar)]` are referred to hence the "style" in the title

**Comment by ZuseZ4** [comment]
@sYgbpNA5UEVufjsHNwF3Xvf9ANZi5bNs Both of your issues should be resolved now, but please let me know if you find other ways to break it. Frontend bugs are a great way to get new contributors started, so I always appreciate them.
